lib/plugins/openaps.js

```diff
@@ -23,17 +23,130 @@ function init() {
   };
 
   openaps.setProperties = function setProperties (sbx) {
-    sbx.offerProperty('openaps', function setopenaps ( ) {
-      var openapsStatus = _.findLast(sbx.data.devicestatus, function (status) {
-        return sbx.entryMills(status) <= sbx.time && ('openaps' in status);
-      });
+    sbx.offerProperty('openaps', function setOpenAPS ( ) {
+      return openaps.analyzeData(sbx);
+    });
+  };
 
-      var loopStatus = checkLoopStatus(openapsStatus, sbx);
-      openapsStatus = openapsStatus || {};
-      openapsStatus.status = loopStatus;
+  openaps.analyzeData = function analyzeData (sbx) {
+    var recentHours = 6; //TODO dia*2
+    var recentMills = sbx.time - times.hours(recentHours).msecs;
 
-      return openapsStatus;
+    var recentData = _.filter(sbx.data.devicestatus, function (status) {
+      return ('openaps' in status) && sbx.entryMills(status) <= sbx.time && sbx.entryMills(status) >= recentMills;
     });
+
+    var prefs = openaps.getPrefs(sbx);
+    var recent = moment(sbx.time).subtract(prefs.warn, 'minutes');
+
+    var result = {
+      seenDevices: {}
+      , lastEnacted: null
+      , lastNotEnacted: null
+      , lastSuggested: null
+      , lastIOB: null
+    };
+
+    function getDevice(status) {
+      var uri = status.device || 'device';
+      var device = result.seenDevices[uri];
+
+      if (!device) {
+        device = {
+          name: uri.indexOf('openaps://') == 0 ? uri.substring('openaps://'.length) : uri
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Expected '===' and instead saw '=='.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1940844314/issues/source?bid=2452379&fileBranchId=3088515#l56)